packages/relay-compiler/bin/__tests__/RelayCompilerMain-test.js

```diff
@@ -10,44 +10,144 @@
+  it('throws when schema path does not exist', async () => {
     await expect(
       main({
-        schema: '/nonexisting/schema.graphql',
-        src: '/existing/',
+        schema: './non-existent/schema.graphql',
+        src: '.',
       }),
     ).rejects.toEqual(
-      new Error('--schema path does not exist: /nonexisting/schema.graphql'),
+      new Error(
+        `--schema path does not exist: ${path.resolve(
+          'non-existent/schema.graphql',
+        )}`,
+      ),
     );
   });
 
-  it('should throw error when src path does not exist', async () => {
+  it('throws when src path does not exist', async () => {
     await expect(
       main({
-        schema: '/existing/schema.graphql',
-        src: '/nonexisting/src',
+        schema: testSchemaPath,
+        src: './non-existent/src',
       }),
-    ).rejects.toEqual(new Error('--src path does not exist: /nonexisting/src'));
+    ).rejects.toEqual(
+      new Error(
+        `--src path does not exist: ${path.resolve('non-existent/src')}`,
+      ),
+    );
   });
 
-  it('should throw error when persist-output parent directory does not exist', async () => {
+  it('throws when persist-output parent directory does not exist', async () => {
     await expect(
       main({
-        schema: '/existing/schema.graphql',
-        src: '/existing/src/',
-        persistOutput: '/nonexisting/output/',
+        schema: testSchemaPath,
+        src: '.',
+        persistOutput: './non-existent/output/',
       }),
     ).rejects.toEqual(
-      new Error('--persist-output path does not exist: /nonexisting/output'),
+      new Error(
+        `--persistOutput path does not exist: ${path.resolve(
+          'non-existent/output',
+        )}`,
+      ),
     );
   });
+
+  describe('concerning the codegen runner', () => {
+    const options = {
+      schema: testSchemaPath,
+      language: 'javascript',
+      include: [],
+      exclude: [],
+      src: '.',
+    };
+
+    describe('and its writer configurations', () => {
+      it('configures the language', () => {
+        const codegenRunner = getCodegenRunner({
+          ...options,
+          language: 'javascript',
+        });
+        const config = codegenRunner.writerConfigs.js;
+        expect(codegenRunner.parserConfigs[config.parser]).not.toBeUndefined();
+      });
+
+      it('configures the file writer with custom scalars', () => {
+        const codegenRunner = getCodegenRunner({
+          ...options,
+          customScalars: {URL: 'String'},
+        });
+        const config = codegenRunner.writerConfigs.js;
+        const writeFiles = config.writeFiles;
+        writeFiles({onlyValidate: true});
+        expect(RelayFileWriter.writeAll).toHaveBeenCalledWith(
+          expect.objectContaining({
+            config: expect.objectContaining({customScalars: {URL: 'String'}}),
+          }),
+        );
+      });
+    });
+  });
+
+  describe('concerning language plugins', () => {
```

Comment:
`describe('getLanguagePlugin()', ...)` since that's what this is testing?

Reply:
Ah my bad, I should have followed the patterns in other test files 👍 (I personally most of the times don’t use just symbols as descriptions.)